crates/examples/infra/mod.rs

```diff
@@ -460,20 +582,38 @@ pub trait RunDA<
                             // when we make progress, submit new events
                         }
                         EventType::ReplicaViewTimeout { view_number } => {
+                            failed_num_views += 1;
                             warn!("Timed out as a replicas in view {:?}", view_number);
                         }
                         EventType::NextLeaderViewTimeout { view_number } => {
+                            failed_num_views += 1;
```

Comment:
Do we actually ever emit these events?  I don't think we do?  I think they are left over from a long time ago before the event-based model.  (Though that isn't related to this PR)

Reply:
Yeah I've deleted one EventType, but after taking a close look I also deleted `NextLeaderViewTimeout` since it's never emitted, for others I just keep them. This is also the reason I added `EventType::ViewTimeout` to track failed views, together with `assert!(total_num_views <= (failed_num_views + num_successful_commits + 2))`.